### PR TITLE
Include Dart SDK arch in package info

### PR DESCRIPTION
**Pull request by zanderso**
This PR adds a `dart_sdk_arch` field to the description of a release bundle. This field describes the target arch of the bundled Dart SDK. For non-x64 architectures, this arch string is also included in the filename of the release bundle.

### CODE REVIEW

**Comment by zanderso** [comment]
@christopherfujino Is there any way to test this by packaging a commit on master?

**Comment by zanderso** [comment]
Here's a website PR to go along with this one that displays the information https://github.com/flutter/website/pull/6736

**Comment by zanderso** [comment]
I added unit tests and enabled local testing with commits on master by disabling 'strict' mode when the `--dry-run` flag is passed. This change is ready for review.

**Comment by christopherfujino** [comment]
> @christopherfujino Is there any way to test this by packaging a commit on master?

I'd have to look at a recipe, but if you run it in led it might upload it to an experimental (unused) storage bucket, where you can then download it and verify that the contents are correct.

This change looks good, but it is blocked by https://github.com/flutter/flutter/issues/97536

File: dev/bots/prepare_package.dart [hunk 1]
```diff
@@ -255,6 +264,8 @@ class ArchiveCreator {
   late File _outputFile;
   final Map<String, String> _version = <String, String>{};
   late String _flutter;
+  late String _dart;
+  late String _dartArch;
```
Comment (jmagman) [inline dev/bots/prepare_package.dart:268]: You can make this a `late final`:
```dart
  late final Future<String> _dartArch = () async {
    // Parse 'arch' out of a string like '... "os_arch"\n'.
    return _runDart(<String>['--version']))
        .trim().split(' ').last.replaceAll('"', '').split('_')[1];
  }();
```

File: dev/bots/prepare_package.dart [hunk 2]
```diff
@@ -255,6 +264,8 @@ class ArchiveCreator {
   late File _outputFile;
   final Map<String, String> _version = <String, String>{};
   late String _flutter;
+  late String _dart;
```
Comment (jmagman) [inline dev/bots/prepare_package.dart:267]: This isn't really related to the arch change, but a better pattern than the `late` ivars would probably be a factory constructor:
```dart
  factory ArchiveCreator.archive(
    Directory tempDir,
    Directory outputDir,
    String revision,
    Branch branch, {
    bool strict = true,
    ProcessManager? processManager,
    bool subprocessOutput = true,
    Platform platform = const LocalPlatform(),
    HttpReader? httpReader,
  })  {
    final Directory flutterRoot = Directory(path.join(tempDir.path, 'flutter'));
    final ProcessRunner processRunner = ProcessRunner(
      processManager: processManager,
      subprocessOutput: subprocessOutput,
      platform: platform,
    );
    processRunner.environment['PUB_CACHE'] = path.join(flutterRoot.absolute.path, '.pub-cache');

    final String flutterExecutable = path.join(
      flutterRoot.absolute.path,
      'bin',
      'flutter',
    );
    final String dartExecutable = path.join(
      flutterRoot.absolute.path,
      'bin',
      'cache',
      'dart-sdk',
      'bin',
      'dart',
    );
    return ArchiveCreator._(
      tempDir: tempDir,
      platform: platform,
      flutterRoot: flutterRoot,
      outputDir: outputDir,
      revision: revision,
      branch: branch,
      strict: strict,
      processRunner: processRunner,
      httpReader: httpReader ?? http.readBytes,
      flutterExecutable: flutterExecutable,
      dartExecutable: dartExecutable,
    );
  }
```
```dart
  ArchiveCreator._({
    required this.tempDir,
    required this.platform,
    required this.flutterRoot,
    required this.outputDir,
    required this.revision,
    required this.branch,
    required this.strict,
    required ProcessRunner processRunner,
    required this.httpReader,
    required String flutterExecutable,
    required String dartExecutable,
  })  : assert(revision.length == 40),
        _processRunner = processRunner,
        _flutter = flutterExecutable,
        _dart = dartExecutable;
```

File: dev/bots/prepare_package.dart [hunk 3]
```diff
@@ -255,6 +264,8 @@ class ArchiveCreator {
   late File _outputFile;
```
Comment (jmagman) [inline dev/bots/prepare_package.dart:264]: Not related to this change, but this ivar can be removed, it's only used in `createArchive()` and is never read again.
```dart
  Future<File> createArchive() async {
    assert(_version.isNotEmpty, 'Must run initializeRepo before createArchive');
    final File outputFile = File(path.join(outputDir.absolute.path, _archiveName));
    await _installMinGitIfNeeded();
    await _populateCaches();
    await _validate();
    await _archiveFiles(outputFile);
    return outputFile;
  }
```

I could make another PR to clean this up though.

**Review by christopherfujino — APPROVED** [review]
LGTM

File: dev/bots/prepare_package.dart (hunk 1 above)
Comment (zanderso) [reply dev/bots/prepare_package.dart]: Done.

File: dev/bots/prepare_package.dart (hunk 2 above)
Comment (zanderso) [reply dev/bots/prepare_package.dart:298]: Good suggestion. Done.

File: dev/bots/prepare_package.dart (hunk 3 above)
Comment (zanderso) [reply dev/bots/prepare_package.dart]: Good catch. Done.

**Pull request merged**